library/dnet.c

```diff
@@ -791,9 +791,6 @@ int dnet_server_send_write(struct dnet_server_send_ctl *send,
 	dnet_trans_create_send_all(s, &ctl);
 	dnet_session_destroy(s);
 
-	if (!err)
-		err = send->write_error;
-
```

Comment:
But if `dnet_server_send_write` doesn't return error, iterator will not stop. And `write_error` contains that kind of errors which stop iterator. And 2 errors correspond to write error and iterator error, since it will be stopped.